src/lib/swapper/LifiSwapper/LifiSwapper.ts

```diff
@@ -0,0 +1,150 @@
+import type { ChainId as LifiChainId, ChainKey, ConfigUpdate, Token } from '@lifi/sdk'
+import LIFI from '@lifi/sdk'
+import type { Asset } from '@shapeshiftoss/asset-service'
+import type { AssetId, ChainId } from '@shapeshiftoss/caip'
+import { fromChainId } from '@shapeshiftoss/caip'
+import type { EvmChainId } from '@shapeshiftoss/chain-adapters'
+import { evmChainIds } from '@shapeshiftoss/chain-adapters'
+import type {
+  ApprovalNeededOutput,
+  BuildTradeInput,
+  BuyAssetBySellIdInput,
+  ExecuteTradeInput,
+  GetEvmTradeQuoteInput,
+  Swapper,
+  SwapperType,
+  Trade,
+  TradeQuote,
+  TradeResult,
+  TradeTxs,
+} from '@shapeshiftoss/swapper'
+import { SwapError } from '@shapeshiftoss/swapper'
+import { filterAssetIdsBySellable } from 'lib/swapper/LifiSwapper/filterAssetIdsBySellable/filterAssetIdsBySellable'
+import { filterBuyAssetsBySellAssetId } from 'lib/swapper/LifiSwapper/filterBuyAssetsBySellAssetId/filterBuyAssetsBySellAssetId'
+import { getTradeQuote } from 'lib/swapper/LifiSwapper/getTradeQuote/getTradeQuote'
+import { getUsdRate } from 'lib/swapper/LifiSwapper/getUsdRate/getUsdRate'
+import { SWAPPER_NAME, SWAPPER_TYPE } from 'lib/swapper/LifiSwapper/utils/constants'
+
+export class LifiSwapper implements Swapper<EvmChainId> {
+  readonly name = SWAPPER_NAME
+  private readonly lifi: LIFI
+  private chainMap: Map<number, ChainKey> = new Map()
```

Comment:
This would make it easier to grasp what this mapping is - we're keying by a subset of `chainReferences` / `EvmChainId`s
```suggestion
  private chainMap: Map<LifiChainId, ChainKey> = new Map()
```